src/controller/user.ts

```diff
@@ -148,4 +151,167 @@ export default class UserController {
+        const token = jwt.sign({ sub: user.id, role: user.role }, config.jwtSecret, { expiresIn: "30d" })
+        ctx.status = 200
+        ctx.body = { token: token }
+      }
+    }
+  }
+
+  public static async getProfile(ctx: Context): Promise<void> {
+    const userRepository: Repository<User> = getManager().getRepository(User)
+    // try to find user
+    const user: User = await userRepository.findOne({
+      id: ctx.state.user.sub,
+    })
+    if (!user) {
+      ctx.status = 401
+      ctx.body = "User not found"
+    } else {
+      // return user and delete password for safety
+      delete user.password
+      ctx.status = 200
+      ctx.body = user
+    }
+  }
+
+  public static async sendPasswordReset(ctx: Context): Promise<void> {
+    const userRepository: Repository<User> = getManager().getRepository(User)
+    const tokenRepository: Repository<PasswordResetToken> = getManager().getRepository(PasswordResetToken)
+
+    // build up user entity to send password reset email
+    const userToResetPasswordFor: User = new User()
+    userToResetPasswordFor.email = ctx.request.body.email
+
+    // validate user entity for sending the password reset request
+    const errors: ValidationError[] = await validate(userToResetPasswordFor, {
+      groups: ["send-reset"],
+      validationError: { target: false },
+    })
+    if (errors.length > 0) {
+      // return BAD REQUEST status code and errors array
+      ctx.status = 400
+      ctx.body = errors
+    } else {
+      const user: User = await userRepository.findOne(
+        { email: userToResetPasswordFor.email },
+        { relations: ["resetToken"] },
+      )
+      if (!user) {
+        ctx.status = 401
+        ctx.body = "User not found"
+      } else if (!user.verified) {
+        ctx.status = 401
+        ctx.body = "Specified User has not been verified yet"
+      } else if (user.resetToken && user.resetToken.createdAt.getTime() + 60 * 1000 > new Date().getTime()) {
+        // return TOO MANY REQUESTS status code
+        ctx.status = 429
+        ctx.body = "You have to wait before you can request another password reset"
+      } else {
+        if (user.resetToken) {
+          await tokenRepository.remove(user.resetToken)
+        }
+        // generate new token to reset the password with
+        const tokenToBeSaved: PasswordResetToken = new PasswordResetToken()
+        tokenToBeSaved.token = crypto.randomBytes(6).toString("hex")
+        tokenToBeSaved.user = user
+        await tokenRepository.save(tokenToBeSaved)
+
+        try {
+          await email.send({
+            template: "login",
+            message: {
+              to: user.email,
+            },
+            locals: {
+              uname: user.username,
+              token: tokenToBeSaved.token,
+              rurl: config.baseUrl + "users/reset/",
+            },
+          })
+        } catch (err) {
+          console.log(err)
+          ctx.throw(500, "Could not send email")
```

Comment:
Die Logik würde ich in Mail-Service-Classe auslagern, da du sicher oft noch Mails senden wirst. Alleine schon was User angeht, fallen mir 3 unterschiedliche Fälle ein. Daher, kann man schon von Anfang an hier ein Service dafür bauen. Macht auch die Funktion hier deutlich übersichtlicher.